apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -138,14 +136,16 @@ export const publishModel = createAsyncThunk(
 export const saveModelCard = createAsyncThunk(
   'aichat/saveModelCard',
   async (_, thunkAPI) => {
-    const rootState = (await thunkAPI.getState()) as RootState;
-    const {currentAiCustomizations, savedAiCustomizations} = rootState.aichat;
     const {dispatch} = thunkAPI;
-
-    const {modelCardInfo} = currentAiCustomizations;
+    const modelCardInfo = await (thunkAPI.getState() as RootState).aichat
+      .currentAiCustomizations.modelCardInfo;
     if (!hasFilledOutModelCard(modelCardInfo)) {
-      dispatch(setHasPublished(false));
+      dispatch(setModelCardProperty({property: 'isPublished', value: false}));
     }
+
+    const {currentAiCustomizations, savedAiCustomizations} = await (
```

Comment:
Nit: I see you call on `getState` twice here, once for `modelCardInfo` in `currentAiCustomizations`, then to retrieve `currentAiCustomizations` and `saved...`. Could call once on `getState` once - but this may be cleaner?

Reply:
Yeah I had to split it up because we want to grab the state to save to S3 after we've potentially updated the `isPublished` state.